tests/Doctrine/Tests/ORM/Functional/SchemaTool/DDC3460Test.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace Doctrine\Tests\ORM\Functional\SchemaTool;
+
+use Doctrine\ORM\Tools;
+
+class DDC3460Test extends \Doctrine\Tests\OrmFunctionalTestCase
+{
+    /**
+     * @group DDC-3460
+     */
+    public function testDetectMyISAM()
+    {
+        $conn = $this->_em->getConnection();
+
+        if (strpos($conn->getDriver()->getName(), "mysql") === false) {
+            $this->markTestSkipped('this test is only relevant for MySQL');
+        }
+
+        $conn->exec('CREATE TABLE `tweet_user` (id INT NOT NULL AUTO_INCREMENT, name VARCHAR(255) NOT NULL, PRIMARY KEY(id)) ENGINE = MyISAM');
+        $conn->exec('CREATE TABLE `tweet_tweet` (id INT NOT NULL AUTO_INCREMENT, content VARCHAR(255) NOT NULL, author_id INT, PRIMARY KEY(id)) ENGINE = MyISAM');
+
+        $fromSchema = $conn->getSchemaManager()->createSchema();
+
+        $classMetadata = [
+            $this->_em->getClassMetadata('Doctrine\Tests\Models\Tweet\Tweet'),
```

Comment:
I was kind of hoping that this might get backported to 2.5 eventually, so I refrained from using PHP5.5-only stuff. But I can change it ofc